pkg/token/interactive.go

```diff
@@ -51,32 +54,55 @@ func (p *InteractiveToken) Token() (adal.Token, error) {
 	clientOpts := azcore.ClientOptions{Cloud: cloud.Configuration{
 		ActiveDirectoryAuthorityHost: authorityFromConfig.String(),
 	}}
-	cred, err := azidentity.NewInteractiveBrowserCredential(&azidentity.InteractiveBrowserCredentialOptions{
-		ClientOptions: clientOpts,
-		TenantID:      p.tenantID,
-		ClientID:      p.clientID,
-	})
-	if err != nil {
-		return emptyToken, fmt.Errorf("unable to create credential. Received: %w", err)
-	}
+	scopes := []string{p.resourceID + "/.default"}
 
-	// Use the token provider to get a new token
-	interactiveToken, err := cred.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: []string{p.resourceID + "/.default"}})
-	if err != nil {
-		return emptyToken, fmt.Errorf("expected an empty error but received: %w", err)
-	}
-	if interactiveToken.Token == "" {
-		return emptyToken, errors.New("did not receive a token")
+	var token string
+	var expirationTimeUnix int64
+	var err error
+
+	if len(p.popClaims) > 0 {
+		// If PoP token support is enabled and the correct u-claim is provided, use the MSAL
+		// token provider to acquire a new token
+		token, expirationTimeUnix, err = pop.AcquirePoPTokenInteractive(
+			context.Background(),
```

Comment:
Usually this is a good practice to start a context as soon as we can in a code and make it flow (usefull for tracing for example). 

A bit out of scope of that PR, but we should at least declare a `ctx := context.Background()` at the top for the function and re-use it.

Reply:
Good point, updated to use a single ctx for all the calls in this func